ICAFIX/ApplyAutoRecleanPipeline.sh

```diff
@@ -0,0 +1,209 @@
+opts_AddOptional '--model-folder' 'ModelFolder' 'string' "the folder path of the trained models" "$HCPPIPEDIR/ICAFIX/rclean_models"
+opts_AddOptional '--model-to-use' 'ModelToUse' 'string' "the models to use separated by '@'" "RandomForest@MLP"
+opts_AddOptional '--vote-threshold' 'VoteThresh' 'integer' "a decision threshold for determing reclassifications, should be less than to equal to the number of models to use" ""
+opts_AddOptional '--matlab-run-mode' 'MatlabMode' '0, 1, or 2' "defaults to $g_matlab_default_mode
+
+0 = compiled MATLAB
+1 = interpreted MATLAB
+2 = Octave" "$g_matlab_default_mode"
+opts_ParseArguments "$@"
+
+#display the parsed/default values
+opts_ShowValues
+
+if ((pipedirguessed))
+then
+    log_Err_Abort "HCPPIPEDIR is not set, you must first source your edited copy of Examples/Scripts/SetUpHCPPipeline.sh"
+fi
+
+case "$MatlabMode" in
+    (0)
+        if [[ "${MATLAB_COMPILER_RUNTIME:-}" == "" ]]
+        then
+            log_Err_Abort "to use compiled matlab, you must set and export the variable MATLAB_COMPILER_RUNTIME"
+        fi
+        ;;
+    (1)
+        #NOTE: figure() is required by the spectra option, and -nojvm prevents using figure()
+        matlab_interpreter=(matlab -nodisplay -nosplash)
+        ;;
+    (2)
+        matlab_interpreter=(octave-cli -q --no-window-system)
+        ;;
+    (*)
+        log_Err_Abort "unrecognized matlab mode '$MatlabMode', use 0, 1, or 2"
+        ;;
+esac
+
+ReclassifyAsSignalFile="ReclassifyAsSignalRecleanVote${VoteThresh}.txt"
+ReclassifyAsNoiseFile="ReclassifyAsNoiseRecleanVote${VoteThresh}.txt"
+
+# compute addtional features, inference by new base learners, produce reclassify files
+log_Msg "Begin to run the reclean pipeline..."
+"$HCPPIPEDIR"/ICAFIX/scripts/RecleanClassify.sh \
+    --study-folder="$StudyFolder" \
+    --subject="$Subject" \
+    --fmri-names="$fMRINames" \
+    --mrfix-concat-name="$MRFixConcatName" \
+    --fix-high-pass="$HighPass" \
+    --fmri-resolution="$fMRIResolution" \
+    --subject-expected-timepoints="$subjectExpectedTimepoints" \
+    --surf-reg-name="MSMAll" \
+    --low-res="${LowResMesh}" \
+    --python-singularity="${PythonSingularity}" \
+    --python-singularity-mount-path="${PythonSingularityMountPath}" \
+    --python-interpreter="${PythonInterpreter}" \
+    --model-to-use="$ModelToUse" \
+    --vote-threshold="$VoteThresh" \
+    --reclassify-as-signal-file="$ReclassifyAsSignalFile" \
+    --reclassify-as-noise-file="$ReclassifyAsNoiseFile"
+
+# apply the reclassification results
+log_Msg "Begin to apply the reclassification results..."
+IFS='@' read -a fMRINamesArray <<<"$fMRINames"
+
+# check SR or MR FIX
+if [ ! ${MRFixConcatName} = "" ] ; then
+    fMRINameToUse=${MRFixConcatName}
+else
+    fMRINameToUse=""
+    for fMRIName in "${fMRINamesArray[@]}"
+    do
+        fMRINameToUse+=" $fMRIName"
+    done
+    # Remove the leading space
+    fMRINameToUse="${fMRINameToUse:1}"
+fi
+
+for fMRIName in ${fMRINameToUse} ; do
```

Comment:
```suggestion
if [[ "$MRFixConcatName" != "" ]]
then
    fMRINamesToUse=("$MRFixConcatName")
else
    fMRINamesToUse=("${fMRINamesArray[@]}")
fi

for fMRIName in "${fMRINamesToUse[@]}" ; do
```

Reply:
Additionally, the only thing between this and the next test for MRFix mode is a few hardcoded params and a log, so instead, I would consider adding an ApplyHand... call before each ReApply... call, (for SR FIX, inside that existing loop), and remove the "apply reclassification" log message above.